addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -795,16 +802,19 @@ var SnippetsMenu = Widget.extend({
         }
 
         if ($selectorSiblings) {
-            $selectorSiblings.filter(':not(.oe_drop_zone):not(.oe_drop_clone)').each(function () {
+            $selectorSiblings.filter(':not(.oe_drop_zone)').each(function () {
```

Comment:
I did not understand your code so I investigated a bit. I wondered why you were not adding the dropzone after the drop clone in all cases as we talked about then just change the following condition into `$zone.prev('.oe_drop_zone:visible, .oe_drop_clone').length === 0`. In fact it is more complicate than I thought but your PR does not solve alll buggy cases and works by chance in the case we have.

What you do here is with this change is "Oh in fact we should not ignore the oe_drop_clone element in the siblings stuff... but treat is as a separate case etc". This will not solve the fundamental problem of: if there is a clone at a location, there *must* be a zone after it **whatever the siblings/children rules are**. Here your fix will solve the "unique column in banner" problem but by chance: the oe_drop_clone was found by chance by the siblings rules. But this is pure chance: the rule is `data-selector=".row > div" data-drop-near=".row > div"` (= the div which are children of rows can be put near other div which are children of rows). If the rule was written as `data-selector=".row > [class*=col]" data-drop-near=".row > [class*=col]"`... your fix would not solve the problem as your clone would not match any rule any more.

Also re-reading this code, I found another problem: the rule above could be written like that `data-selector=".row > div" data-drop-in=".row"` (the only difference would be that columns would now be possible to drop in empty rows (maybe we could want that but that's another debate)). But if you write it like that your don't have the same visual:
![Screenshot from 2020-11-12 17-00-34](https://user-images.githubusercontent.com/10338094/98964205-d47f2300-2508-11eb-969b-5f717e542662.png)
Instead of the wanted:
![Screenshot from 2020-11-12 17-01-05](https://user-images.githubusercontent.com/10338094/98964232-dcd75e00-2508-11eb-91be-e2664d6cf2e9.png)
(who knows maybe the rule was written data-drop-near=".row > div" to avoid that problem...)
This is because the first and last drop zones are handled by the "children rules" code in that case... but this code should yield the same result: vertical dropzone.

I think the code should be reorganized:
1. Add a dropzone after the clone whatever the rules are
    - Determine if it must be vertical or horizontal by checking neighbors: the previous one if there is one or the next one other wise
    - If no neighbords, use the same condition as for the children rules to determine if it must be vertical or not
2. Review children/siblings dropzone creation to handle the bug I mentioned above.
3. If (2) is done properly I am not sure the final cleanup at the end of this function will still be necessary

Reply:
@qsm-odoo "If no neighbords, use the same condition as for the children rules to determine if it must be vertical or not"
You mean what rule ? I think that if no neighbord, the dropzone could be always horizontal.. what do you think ? Because otherwise I need to know the height of the dragged element (and if it is full width or not) But does it matter ?

Reply:
I mean the rule which is in the code to refactore: https://github.com/odoo/odoo/blob/12.0/addons/web_editor/static/src/js/editor/snippets.editor.js#L757

Reply:
@qsm-odoo about the point 3, did you found a user case where the the final cleanup is usefull ? Because I would like to test I we still need it after my fixes

Reply:
I don't have an example no. Just keep the cleanup in 12.0 anyway, who knows what the community did. We'll remove it in master after some random testing.